### ISSUE TITLE
No host IP error

### ISSUE THREAD

**Issue by justicel** [Learnosity/vagrant-nfs_guest#12]
Hello, I'm trying out your gem but am unable to use it so far. I get the following error:

No host IP was given to the Vagrant core NFS helper. This is
an internal error that should be reported as a bug.

I saw an issue report about this previously but it didn't seem like there was a whole lot of resolution to it. Any ideas on how to resolve this? I'm using the latest gem version.

**Comment by justicel** [comment]
So, I've narrowed the issue/error down. It appears to be due to the vagrant-parallels plugin (Using parallels instead of virtualbox). In theory all functions should be supported and passed through but maybe the plugin is sending the data through in a way that you're not expecting. Anything I can do to help figure this out further?

**Comment by alangarf** [comment]
Try is setting static IP addresses, we recently had a PR that added support for this, and apparently allowed VMWare to work. This plugin relies on VirtualBox methods to retrieve the IP addresses and this is likely the cause.

I'll look into the parallels plugin to see if there is a similar method available to get the IPs.

Thanks.

**Comment by justicel** [comment]
Okay thanks. What is the syntax to set the IP info then? I didn't see any
documentation for it. Thanks!

**Comment by alangarf** [comment]
Actually I can see that Vagrant has been updated to fix the issue with the nfs_host_ip etc on synced folders, so this may actually be fixable for all. I'll update you when I have found something.

**Comment by justicel** [comment]
Okay, thanks. I just tried using the host_ip and guest_ip settings and unless I'm using them incorrectly I don't think they'll work in this case, so guess I'll need to wait and see if the fixes inside Vagrant work :)

**Comment by alangarf** [comment]
So far all I have is to change the plugin to supercede the built in NFS standard plugin. This works flawlessly but then you have to use NFS Guest for all NFS usages (which I personally don't see this as a concern, but it is a huge shift in the plugins historic usage).

The reason this is a problem is the build in "PrepareNFSSettings" event explicitly checks for the symbol `:nfs` which comes from the `type` given in the VagrantFile. I am currently doing this via `:nfs_guest` type but then I have to implement my own "PrepareNFSSetting" like event for *every* provider.

The simplest solution to have it work for all providers is to supercede the base :nfs type. But I'm not sure how that would go down with the plugin's users.

Ideas?

**Comment by dmatora** [comment]
well i guess it would be nice to have it as an option, so that it doesn't turn off users who have many project and don't want to fix them all
personally, i don't care about original nfs, i only need it reversed

**Comment by alangarf** [comment]
I don't want to override the internal `PrepareNFSSettings` in Vagrant itself to handle `:nfs_guest`. This is problematic as the plugin then has to updated any time Vagrant changes. So supporting all the providers in the plugin itself is probably the only option then. :(

**Comment by dmatora** [comment]
you can have something like force_original_keyword option,
when it's enabled, plugin can expect :nfs
and when it is disabled, plugin can expect :nfs_guest
this way both kind of users can be happy

**Comment by alangarf** [comment]
That's actually a good idea. I'll see what I can do. Thanks!

**Comment by dmatora** [comment]
my pleasure :)

**Comment by dmatora** [comment]
i just realized, that it works fine in parallels with
```
  config.vm.network :private_network, ip: "192.168.5.55"
  config.nfs_guest.host_ip = "192.168.5.1"
  config.nfs_guest.guest_ip = "192.168.5.55"
```
obviously not a solution for public_network (bridged), but still nice that we don't have to mess with original nfs for private_network

**Comment by alangarf** [comment]
Hey guys,

Here is a PR with the changes I'm proposing to make it work with all providers by replacing the standard NFS plugin. I still have to add the "force_original" option, but it would be good to test it.

https://github.com/Learnosity/vagrant-nfs_guest/pull/13

**Comment by alangarf** [comment]
@justicel @dmatora hopefully this should all be fixed in v0.1.7!
